From 318b151a21bd5558801e6af939e8a7f0606026ec Mon Sep 17 00:00:00 2001
From: Larry Woodman <lwoodman@redhat.com>
Date: Tue, 18 Nov 2014 19:49:49 -0500
Subject: [mm] vmstat: create fold_diff

Message-id: <1416340191-18643-3-git-send-email-lwoodman@redhat.com>
Patchwork-id: 100253
O-Subject: [RHEL7.1 PATCH 2/4] vmstat: create fold_diff
Bugzilla: 1157802
RH-Acked-by: Rik van Riel <riel@redhat.com>
RH-Acked-by: Rafael Aquini <aquini@redhat.com>
RH-Acked-by: Motohiro Kosaki <mkosaki@redhat.com>

commit 4edb0748b23887140578d68f5f4e6e2de337a481
 Author: Christoph Lameter <cl@linux.com>
 Date:   Wed Sep 11 14:21:31 2013 -0700

    vmstat: create fold_diff

    Both functions that update global counters use the same mechanism.

    Create a function that contains the common code.

    Signed-off-by: Christoph Lameter <cl@linux.com>
    Cc: KOSAKI Motohiro <kosaki.motohiro@jp.fujitsu.com>
    CC: Tejun Heo <tj@kernel.org>
    Cc: Joonsoo Kim <js1304@gmail.com>
    Cc: Alexey Dobriyan <adobriyan@gmail.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/mm/vmstat.c b/mm/vmstat.c
index 750482b..3f30142 100644
--- a/mm/vmstat.c
+++ b/mm/vmstat.c
@@ -414,6 +414,15 @@ void dec_zone_page_state(struct page *page, enum zone_stat_item item)
 EXPORT_SYMBOL(dec_zone_page_state);
 #endif
 
+static inline void fold_diff(int *diff)
+{
+ int i;
+
+ for (i = 0; i < NR_VM_ZONE_STAT_ITEMS; i++)
+  if (diff[i])
+   atomic_long_add(diff[i], &vm_stat[i]);
+}
+
 /*
  * Update the zone counters for the current cpu.
  *
@@ -483,10 +492,7 @@ static void refresh_cpu_vm_stats(int cpu)
    drain_zone_pages(zone, &p->pcp);
 #endif
  }
-
- for (i = 0; i < NR_VM_ZONE_STAT_ITEMS; i++)
-  if (global_diff[i])
-   atomic_long_add(global_diff[i], &vm_stat[i]);
+ fold_diff(global_diff);
 }
 
 /*
@@ -516,9 +522,7 @@ void cpu_vm_stats_fold(int cpu)
    }
  }
 
- for (i = 0; i < NR_VM_ZONE_STAT_ITEMS; i++)
-  if (global_diff[i])
-   atomic_long_add(global_diff[i], &vm_stat[i]);
+ fold_diff(global_diff);
 }
 
 /*
-- 
1.7.1